Gh-465: Update gaffer gremlin docs for 2.2 (#466)
gchq/gaffer-doc: docs/user-guide/query/gremlin/gremlin.md

# Gremlin in Gaffer

!!! warning
    GafferPop is still under development and has some [limitations](gremlin-limits.md).
    The implementation may not allow some advanced features of Gremlin and it's
    performance is unknown in comparison to standard Gaffer `OperationChains`.

[Gremlin](https://tinkerpop.apache.org/gremlin.html) is a query language for
traversing graphs. It is a core component of the Apache Tinkerpop library and
allows users to easily express more complex graph queries.

GafferPop is a lightweight Gaffer implementation of the [TinkerPop framework](https://tinkerpop.apache.org/),
where TinkerPop methods are delegated to Gaffer graph operations.

The addition of Gremlin as query language in Gaffer allows users to represent
complex graph queries in a simpler language akin to other querying languages
used in traditional and NoSQL databases. It also has wide support for various
languages so for example, you can write queries in Python via the [`gremlinpython` library](https://pypi.org/project/gremlinpython/)

!!! tip

>     in-depth tutorials on Gremlin as a query language, its associated libraries

Reviewer: ```suggestion
    In-depth tutorials on Gremlin as a query language and its associated libraries
```

    can be found in the [Apache Tinkerpop Gremlin docs](https://tinkerpop.apache.org/gremlin.html).

## Using Gremlin Queries in Gaffer

> Gremlin was added to Gaffer as a new graph query language in version 2.1 and since

Reviewer: ```suggestion
Gremlin was added to Gaffer in version 2.1 as a new graph query language and since
```

version 2.2 a container image is provided allowing a Gremlin layer to be added to
existing 2.x graphs. A full tutorial on setting up this image is provided in the
[administration guide](../../../administration-guide/gaffer-deployment/gremlin.md).

This guide will use the [Python API for Gremlin](https://pypi.org/project/gremlinpython/)
to demonstrate some basic capabilities and how they compare to standard Gaffer syntax.

> To start querying in Gremlin we first need a reference to whats know as the

Reviewer: ```suggestion
To start querying in Gremlin we first need a reference to what is known as the
```

> Graph Traversal. To obtain this we need to connect to a running Gremlin server

Reviewer: ```suggestion
Graph Traversal. To obtain this we need to connect to a running Gremlin server,
```
